content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,241 @@
+branch. The pull request is mergebale or not can also be decided by the payload of webhook also.
+
+https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+
+=== How to run the Plugin
+See https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+and set credentials, owner and repository on your own and you will be good to go.
+
+=== What is avalilable now
+1. User will be able to select to use Jenkinsfile.yaml for build.
+2. Git Push step
+3. harvest results and reports [and post in the PR]
+ a. junit()
+ b. findbugs()
+ c. archiveArtifacts()
+4. Basic interface to parse and get build specifications from YAML file.
+
+=== Things decided
+1. To build the plugin on the top of multibranch pipeline plugin. As that plugin has implementation of
+  a. Nice interface to show diffrent branch and pull requests build seperately with use of suitable plugins like Github, Bitbucket.
+  b. Detect trusted revisions in a repository.
+  c. Pblishing of build status to the repository.
+
+2. Convert the YAML configuration to declarative pipeline.
+3. User will provide path to the script relative to the root directory of the repository
+  without extension (.sh or .bat) in the YAML file. The plugin will generate pipeline script to detect the
+  platform and call .sh or .bat script.
+
+  Example:
+    Path provided: ./scripts/hello
+    a. On UNIX machine “./scripts/hello.sh” will be called
+    b. On non-UNIX machine “./scripts/hello.bat” will be called.
+
+=== Implementations till now
+
+A first prototype of the plugin is ready. It supports all features of Multi-Branch Pipeline and offers the following features.
+
+Build description is defined via YAML file stored within the SCM repo. This plugin
+will depend on GitHub plugin, Bitbucket plugin, Gitlab plugin if users will be
+using respective paltfroms for their repositories.
+[start=1]
+. Basic conversion of YAML to Declarative Pipeline: A class YamlToPipeline
+is written which will load the "Jenkinsfile.yaml" and make use of PipelineSnippetGenerator class
+to generate Declarative pipeline code.
+. Reporting of results.
+. Plugin is using Yaml from target branch right now. (Maybe this needs some discussion, example: what if PR contains changes in Jenkinsfile.yaml)
+. Git Push step: To push the changes of PR to the target branch. This is implemented
+using git-plugin, PushCommand is used for this from git-plugin. credentialId,
+branch name and repository url for intracting with Github, Bitbucket, etc
+will be taken automatically from respective plugins plugin. (Yes users have to
+configure in that way, you can see https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo])
+
+=== Jenkinsfile.yaml exapmle
+Write now no format has been decided for the yaml file. But for thes demo I am using below yaml Jenkinsfile
```

Comment:
typo: thes

and I think `am using below yaml Jenkinsfile` should `Jenkinsfile.yaml`, no?